java/vector/src/main/java/org/apache/arrow/vector/complex/BaseRepeatedValueVector.java

```diff
@@ -186,6 +186,12 @@ public void clear() {
     super.clear();
   }
 
+  @Override
+  public void reset() {
+    vector.reset();
```

Comment:
I believe we should reset the offset vector too.

Reply:
zero out the buffer like this right?
`offsetBuffer.setZero(0, offsetBuffer.capacity());`

Reply:
yes

Reply:
also should also zero out validity buffers and any others right?

Reply:
the reset on the inner data vector will be taken care of by it's invocation. here we just need to reset the offset and validity buffer of the list vector.